### PR TITLE
Start test without time limit

### PR DESCRIPTION
**Pull request by tranthuc141**
Start test without time limit

### CODE REVIEW

**Comment by nhieu92** [comment]
Ok !

File: app/controllers/exams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @exam = Exam.new exam_params
+    if @exam.save
+      redirect_to exams_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @exam = Exam.new
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove

File: app/controllers/exams_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @exams = current_user.exams
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove ()

File: app/controllers/exams_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @checked_exams = @exams.where status: 'checked'
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove

File: app/controllers/exams_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    @ready_exams = @exams.where status: 'not_check_yet'
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove

File: app/controllers/results_controller.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+class ResultsController < ApplicationController
+  before_action :authenticate_user!
+  skip_before_action :verify_authenticity_token
+
+  def new
```
Comment (dieunb) [inline app/controllers/results_controller.rb]: remove this function

File: app/controllers/results_controller.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    @result = Result.new
+    @exam = Exam.find params[:exam_id]
+    @result.answers.build
+  end
+
+  def create
+    @result = Result.new result_params
+    if @result.save
+      @result.exam.update_attributes status: 'checked'
```
Comment (dieunb) [inline app/controllers/results_controller.rb]: move this to callback

File: app/controllers/results_controller.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      redirect_to exams_path
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @result = Result.find params[:id]
+    @course = @result.exam.course
```
Comment (dieunb) [inline app/controllers/results_controller.rb]: remove

File: app/controllers/results_controller.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    @answers = @result.answers
```
Comment (dieunb) [inline app/controllers/results_controller.rb]: remove

File: app/models/exam.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+class Exam < ActiveRecord::Base
+  has_one :result
+  belongs_to :user
+  belongs_to :course
+
+  scope :ready_exams, -> {
+    where status: 'not_check_yet'
+  }
+
+  scope :checked_exams, -> {
```
Comment (dieunb) [inline app/models/exam.rb]: not need, remove

File: app/models/exam.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+class Exam < ActiveRecord::Base
+  has_one :result
+  belongs_to :user
+  belongs_to :course
+
+  scope :ready_exams, -> {
```
Comment (dieunb) [inline app/models/exam.rb]: not need remove

Comment (dieunb) [inline app/models/exam.rb]: write on a line

File: app/models/exam.rb (hunk 9 above)
Comment (dieunb) [inline app/models/exam.rb]: write on a line

File: app/controllers/exams_controller.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
+class ExamsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @exams = current_user.exams
+    @checked_exams = @exams.checked_exams
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove

File: app/controllers/exams_controller.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    @ready_exams = @exams.ready_exams
```
Comment (dieunb) [inline app/controllers/exams_controller.rb]: remove, call it on views

File: config/routes.rb [hunk 13]
```diff
@@ -4,6 +4,10 @@
 
   resources :users
   resources :courses
+  resources :results
```
Comment (dieunb) [inline config/routes.rb]: remove

File: config/routes.rb [hunk 14]
```diff
@@ -4,6 +4,10 @@
 
   resources :users
   resources :courses
```
Comment (dieunb) [inline config/routes.rb]: remove, not need.

File: app/models/result.rb [hunk 15]
```diff
@@ -1,5 +1,17 @@
...
+  after_create :update_status_exam
+
+  belongs_to :exam
   has_many :answers, dependent: :destroy
+
+  accepts_nested_attributes_for :answers
+
+  def get_score
+    answers = self.answers
+    answers.select{|a| a.option.correct}.count
```
Comment (dieunb) [inline app/models/result.rb]:
```
answers.count{|answer| answer.option.correct}

File: app/views/exams/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    <div class="col-md-4">
+      <% @courses.each do |course| %>
+        <div class="checkbox inline">
+          <%= f.check_box :course_id, {}, course.id, false %> <%= course.name %> <br />
+        </div>
+      <% end %>
+    </div>
+
+    <div class="col-md-4"> 
+      <%= f.hidden_field :user_id, value: current_user.id %>
```
Comment (dieunb) [inline app/views/exams/index.html.erb]: not need, remove. Can call current_user

File: app/models/result.rb [hunk 17]
```diff
@@ -1,5 +1,17 @@
...
+  after_create :update_status_exam
+
+  belongs_to :exam
   has_many :answers, dependent: :destroy
+
+  accepts_nested_attributes_for :answers
+
+  def score
+    answers = self.answers
+    answers.select{|answer| answer.option.correct}.count
```
Comment (dieunb) [inline app/models/result.rb]:
```
answers.count{|answer| answer.option.correct}
```

**Pull request merged**